packages/tweeter/src/TweetFilter/PublishedTweetsFilter.php

```diff
@@ -21,23 +21,25 @@ public function __construct(
     }
 
     /**
-     * @param Tweet[] $allTweets
-     * @return Tweet[]
+     * @param PostTweet[] $postTweets
+     * @return PostTweet[]
      */
-    public function filter(array $allTweets): array
+    public function filter(array $postTweets): array
     {
-        return array_filter($allTweets, fn ($tweet): bool => ! $this->wasTweetPublished($tweet));
-    }
-
-    private function wasTweetPublished(Tweet $postTweet): bool
-    {
-        foreach ($this->getPublishedPostTweets() as $publishedTweet) {
-            if ($postTweet->isSimilarToPublishedTweet($publishedTweet)) {
-                return true;
+        $publishedTweets = $this->getPublishedPostTweets();
+
+        return array_filter(
+            $postTweets,
+            function (PostTweet $postTweet) use ($publishedTweets): bool {
+                foreach ($publishedTweets as $publishedTweet) {
+                    if ($publishedTweet->getLink() === $postTweet->getLink()) {
```

Comment:
The most important change is that now we do not compare string, but linked post urls